inc/azure_c_util/containing_record.h

```diff
@@ -16,6 +16,12 @@
 // Calculate the address of the base of the structure given its type, and an
 // address of a field within the structure.
 //
-#define containingRecord(address, type, field) ((type *)((uintptr_t)(address) - offsetof(type,field)))
+
+#ifdef WIN32
+/*windows.h brings CONTAINING_RECORD*/
+#include "windows.h"
```

Comment:
Any benefit from using in one case one containing record and in another case the other? Looks to me that this is simple enough so that we could just use one (our own).

Reply:
Absolutely no benefit. That's why I switched to having just 1 CONTAINING_RECORD with the casing from windows.h across the repos. If we're in Windows land - use that. If we are in linux land - use the #define. But we spend 99.9999% of our time in Windows land - so let's optimize the common case.

There's a problem with "our own casing" (which I find annoying). Pressing F1 on "our own" doesn't open help. Pressing F1 on something that exists however... will bring me the page with the help.

---
In reply to: [460209663](https://github.com/Azure/azure-c-util/pull/51#discussion_r460209663) [](ancestors = 460209663)